src/event.rs

```diff
@@ -425,6 +448,118 @@ pub enum WindowEvent {
     RedrawRequested,
 }
 
+/// Represents the pointer type of a pointer event.
+///
+/// ## Platform-specific
+///
+/// **Wayland/X11:** Can only identify [`Touch`](Self::Touch), any other pointer is identified as
+/// [`Mouse`](Self::Mouse).
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+pub enum PointerType {
```

Comment:
Splitting this from `PointerSource` does not feel sufficiently motivated, but perhaps it will make sense in the future because we're going to get more data on `PointerSource`?

If the issue is the size of `WindowEvent`, I'd argue we `Box` the `Force` struct instead.

Reply:
If we do keep it, then a naming suggestion could be to use `kind` and `PointerKind` instead of `ty` and `PointerType`.

Reply:
> Splitting this from `PointerSource` does not feel sufficiently motivated, but perhaps it will make sense in the future because we're going to get more data on `PointerSource`?

That was the reason indeed.

> If we do keep it, then a naming suggestion could be to use `kind` and `PointerKind` instead of `ty` and `PointerType`.

Yeah, this is way better indeed.
Will adapt it!

Reply:
On another thought, this is also how Web works, another reason to switch to this new event.

Reply:
Done!